assignments/lambda-classes.js

```diff
@@ -1 +1,180 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor(person) {
+        this.name = person.name;
+        this.age = person.age;
+        this.location = person.location;
+        this.gender = person.gender;
+    }
+    speak() {
+       return `Hello my name is ${this.name}, I am from ${this.location}`;
+    }
+}
+
+class Instructor extends Person {
+    constructor(instructor) {
+        super(instructor);
+        this.specialty = instructor.specialty;
+        this.favLanguage = instructor.favLanguage;
+        this.catchPhrase = instructor.catchPhrase;
+        this.gradedAnswer = Instructor.gradedAnswer;
+    }
+    demo(subject) {
+       return `Today we are learning about ${subject}`;
+    }
+
+    grade(student, subject) {
+       return `${student.name} receives a perfect score on ${subject}`;
+    }
+
+    gradedNumber(student, max){
+        student.gradedAnswer = Math.floor(Math.random() * Math.floor(max));
+        return `${student.name}'s score is: ${student.gradedAnswer}`;
+    }
+
+    graduate(student) {
+        if (student.gradedAnswer > 70) {
+            return `${student.name} has graduated!`;
+        } else {
+            return `Try again next time!`;
```

Comment:
Think about how you can loop here until the student scores above 70. What kind of loop will it be?